src/mountscript.sh

```diff
@@ -602,6 +644,324 @@ ensure_aznfswatchdog()
     fi
 }
 
+find_next_available_port_and_start_stunnel()
+{
+    while true
+    do
+        # get the next available port
+        available_port=$(get_next_available_port)
+        if [ $? -ne 0 ]; then
+            eecho "Failed to get the next available port for nfsv4 mount."
+            return 1
+        fi
+        vecho "Next Available Port: '$available_port'"
+
+        if [ -z "$available_port" ]; then
+            eecho "Running out of ports. For Nfsv4, stunnel uses port range from $NFSV4_PORT_RANGE_START to $NFSV4_PORT_RANGE_END. All ports from this range are used by other processes."
+            return 1
+        fi
+
+        used_port=$(cat $stunnel_conf_file | grep accept | cut -d: -f2)
+        vecho "used port: '$used_port'"
+
+        chattr -f -i $stunnel_conf_file
+
+        sed -i "s/$used_port/$available_port/" $stunnel_conf_file
+        if [ $? -ne 0 ]; then
+            eecho "Failed to replace the port in $stunnel_conf_file."
+            chattr -f +i $stunnel_conf_file
+            return 1
+        fi
+        chattr -f +i $stunnel_conf_file
+
+        new_used_port=$(cat $stunnel_conf_file | grep accept | cut -d: -f2)
+
+        # start the stunnel process
+        stunnel_status=$(stunnel $stunnel_conf_file 2>&1)
+        if [ -n "$stunnel_status" ]; then
+            # TODO: Change nfs_host to storageaccount later after server side changes are implemented and merged into Main.
+            is_binding_error=$(echo $stunnel_status | grep "$LOCALHOST:$new_used_port: Address already in use")
+            if [ -z "$is_binding_error" ]; then
+                eecho "[FATAL] Not able to start stunnel process for '${stunnel_conf_file}'!"
+                return 1
+            fi
+        else
+	        vecho "Found new port '$new_used_port' and restarted stunnel."
+	        break
+        fi
+    done
+}
+
+get_next_available_port()
+{
+    for ((port=NFSV4_PORT_RANGE_START; port<=NFSV4_PORT_RANGE_END; port++))
+    do
+        is_port_available=`netstat -tuapn | grep "$LOCALHOST:$port "`
+        if [ -z "$is_port_available" ]; then
+            break
+        fi
+    done
+
+    if [ $port -le $NFSV4_PORT_RANGE_END ]; then
+        echo "$port"
+    else
+        echo ""
+    fi
+}
+
+
+install_CA_cert()
```

Comment:
we might have to have an option to download or use an existing copy of the certificate

Reply:
Mount helper script first checks if the certificate already exists. If not, then it downloads the certificate.